### ISSUE TITLE
An action bound to the ESC key doesn't execute immediately

### ISSUE THREAD

**Issue by numirias** [gwsw/less#83]
**Bug**
I want to quit less by pressing `ESC`, so I added this to my `~/.lesskey` file:

```
#command
\e quit
```
However, when I now press `ESC`, nothing happens yet. Only when I press the key *twice*, the action triggers and less quits.

**Cause**
[In `command.c:getcc()`](https://github.com/gwsw/less/blob/9ea92b9c38feef60319d2e0fdaecb73584a4ea19/command.c#L919), less scans the command character stream to replace the `kent` (Keypad Enter) sequence with a newline character. In consequence,  any key sequence starting the same way as `kent` is not processed any further yet, until the loop in `getcc_repl()` has concluded that it doesn't match.

Since on my machine, `kent` is `\eOM`, any key sequence starting with `\e` or `\eO` is not handled immediately. So, pressing `\e` doesn't trigger any action until I add another character that breaks the `kent` sequence. 

**Workaround**

Since `kent` gets its value from the termcap entry `@8` ([`screen.c:1212`](https://github.com/gwsw/less/blob/9ea92b9c38feef60319d2e0fdaecb73584a4ea19/screen.c#L1212)), it's possible to force it to another value via e.g. this `.lesskey` entry:

```
#env
LESS_TERMCAP_@8 =
```

However, it would be great to not have to use this workaround. A fix could involve checking if there is a key bind shorter than the replacement sequence, in which case the replacement attempt is skipped and the characters are forwarded immediately.

**Comment by gwsw** [comment]
I'm not sure that assigning a command to ESC is a good idea, even if this issue were addressed. On most terminals, several commonly used keys send sequences beginning with ESC. So less would exit if you pressed any arrow key, ENTER, HOME, END, etc. as well as any mouse movement if you are using --mouse.

**Comment by polluks** [comment]
> I'm not sure that assigning a command to ESC is a good idea, even if this issue were addressed. On most terminals, several commonly used keys send sequences beginning with ESC. So less would exit if you pressed any arrow key, ENTER, HOME, END, etc. as well as any mouse movement if you are using --mouse.

Indeed, however my system uses 8-bit CSI so there would be no ambiguity.

**Comment by gwsw** [comment]
When you say your system "uses 8-bit CSI", do you mean that your terminal sends 0x9B instead of 0x1B as the initial character of an escape sequence? If so, why does your termcap say kent is \eOM? Isn't this a configuration error of some kind?

**Comment by polluks** [comment]
This was a special case, my kent sends a simple CR therefore no escape trouble at all.

**Comment by gwsw** [comment]
I'm sorry, I'm still confused. Are you saying that the ENTER key on your terminal sends '\r' but your terminfo says that it sends a sequence containing an ESC? If so, the solution would be to make your terminfo match your terminal.

**Comment by polluks** [comment]
I cannot distinguish return and enter, but it doesn't matter.

**Comment by bew** [comment]
An alternative I use is:
```
\e\e quit
```
Bind Escape-Escape to quit

(and I did the same for `abort` in the line-edit section, to cancel a search for example)

**Comment by gwsw** [comment]
The original request was to disable the keypad enter key if another command is a prefix of that key's sequence. This is not impossible, but is somewhat nontrivial. I think requiring the user to explicitly override the keypad enter key using ```LESS_TERMCAP_@8``` or ```LESS_TERMINFO_kent``` or directly modifying their terminfo entry is not too onerous and ensures that the user knows what they are doing, rather than just accidentally disabling the enter key by defining a new command, which might cause confusion. I think this request is not a clear improvement and is not worth the effort to implement and support, nor the risk of confusion.

**Comment by allefeld** [comment]
> I'm not sure that assigning a command to ESC is a good idea, even if this issue were addressed. On most terminals, several commonly used keys send sequences beginning with ESC. So less would exit if you pressed any arrow key, ENTER, HOME, END, etc. as well as any mouse movement if you are using --mouse.

There are programs like `fzf` which support using both a single ESC key (to quit) and arrow keys etc. I believe it waits for a short time, and if nothing else comes, just acts on the single ESC. Would it be possible to implement something like this in `less` to enable mapping ESC to quit? This would still interfere with "manual escape sequences" like ESC-v, but personally I don't use them.

The reason for the request is that like the OP I use other programs which quit on ESC, and having to remember quickly which one to use leads to mistakes. I now use `export LESSKEY_CONTENT="\e\e quit"`, which (strangely) makes less quit not on ESC-ESC, but on ESC-ESC-ESC. Better than nothing, but not optimal.
